src/core/dash.h

```diff
@@ -958,6 +964,40 @@ auto DashTable<_Key, _Value, Policy>::Traverse(Cursor curs, Cb&& cb) -> Cursor {
   return Cursor{global_depth_, sid, bid};
 }
 
+template <typename _Key, typename _Value, typename Policy>
+template <typename Cb>
+auto DashTable<_Key, _Value, Policy>::TraverseBuckets(Cursor curs, Cb&& cb) -> Cursor {
+  if (curs.bucket_id() >= SegmentType::kTotalBuckets)  // sanity.
+    return 0;
```

Comment:
Wouldn't this restart the main loop of the caller since `0` means to `start` again and probably we ended up in this situation because of what re returned below ? Why not a DCHECK ? (I know it's WIP I just saw this and thought to ask)

Reply:
returning 0 means iteration finished, we have the same logic in Traverse function